pandas/core/arrays/period.py

```diff
@@ -822,7 +822,14 @@ def to_timestamp(self, freq=None, how: str = "start") -> DatetimeArray:
 
         new_parr = self.asfreq(freq, how=how)
 
-        new_data = libperiod.periodarr_to_dt64arr(new_parr.asi8, base)
+        is_start = how == "S"
+        if is_start:
+            new_data = np.asarray(
```

Comment:
```suggestion
            new_data = new_parr.dt.start_time
```

the list comprehension is going to be prohibitively slow (might be the cause of the current ASV failures). If you can use the vectorized datetimeindex (DTI) methods we have, I think that might fix things

Reply:
Hey thanks @WillAyd, I'll have more time later to look into this, but for now...
`new_parr.dt.start_time` gives `AttributeError: 'PeriodArray' object has no attribute 'dt'`
`new_parr.start_time` gives `RecursionError: maximum recursion depth exceeded while calling a Python object`